ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/ScanResultsToolbar.tsx

```diff
@@ -0,0 +1,180 @@
+                }
+                onClear={() => {
+                    setClusterFilter('');
+                    setSearchFilter((prev) => ({ ...prev, clusterName: '' }));
+                }}
+            />
+        </ToolbarItem>
+    );
+}
+
+type ScanResultsToolbarProps = {
+    numberOfScanResults: number;
+    searchFilter: SearchFilter;
+    setSearchFilter: React.Dispatch<React.SetStateAction<SearchFilter>>;
+} & UsePaginationResult;
+
+function ScanResultsToolbar({
+    numberOfScanResults,
+    searchFilter,
+    setSearchFilter,
+    page,
+    perPage,
+    onSetPage,
+    onPerPageSelect,
+}: ScanResultsToolbarProps) {
+    const [scanFilter, setScanFilter] = useState<string>('');
+    const [profileFilter, setProfileFilter] = useState<string>('');
+    const [clusterFilter, setClusterFilter] = useState<string>('');
+    const [searchType, setSearchType] = useState<string>('Profile');
+
+    const resetSearchFilterKey = (key: keyof SearchFilter) => {
+        setSearchFilter((previousFilter) => ({
+            ...previousFilter,
+            [key]: '',
+        }));
+    };
+
+    function onSearchTypeSelect(_e, selection) {
+        switch (selection) {
+            case 'Profile':
+                resetSearchFilterKey('clusterName');
+                setClusterFilter('');
+                break;
+            case 'Cluster':
+                resetSearchFilterKey('profileName');
+                setProfileFilter('');
+                break;
+            default:
+                break;
+        }
+        setSearchType(selection);
+    }
+
+    return (
+        <Toolbar>
+            <ToolbarContent>
+                <ToolbarItem variant="search-filter">
+                    <SearchInput
+                        aria-label="Filter by scan name"
+                        placeholder="Filter by scan name"
+                        value={scanFilter}
+                        onChange={(_e, selection: string) => setScanFilter(selection)}
+                        onSearch={() => setSearchFilter({ ...searchFilter, scanName: scanFilter })}
+                        onClear={() => {
+                            setScanFilter('');
+                            resetSearchFilterKey('scanName');
+                        }}
+                    />
+                </ToolbarItem>
+                <ToolbarItem variant="separator" />
+                {searchType === 'Profile' ? (
+                    <ProfileFilterInput
+                        profileFilter={profileFilter}
+                        setSearchFilter={setSearchFilter}
+                        setProfileFilter={setProfileFilter}
+                    />
+                ) : (
+                    <ClusterFilterInput
+                        clusterFilter={clusterFilter}
+                        setSearchFilter={setSearchFilter}
+                        setClusterFilter={setClusterFilter}
+                    />
+                )}
```

Comment:
Nit, but if these filter inputs aren't going to really get more complex IMO it is more readable to just inline them here.

Reply:
Updated. I go back and forth on this, I feel like large conditionals in the main render can become hard to read sometimes but other times I feel the same way as you.